src/diffusers/schedulers/scheduling_consistency_models.py

```diff
@@ -0,0 +1,370 @@
+        sigma_data (`float`):
+            The standard deviation of the data distribution, following the EDM paper [2]. This was set to 0.5 in the
+            original implementation, which is also the original value suggested in the EDM paper.
+        s_noise (`float`):
+            The amount of additional noise to counteract loss of detail during sampling. A reasonable range is [1.000,
+            1.011]. This was set to 1.0 in the original implementation.
+        rho (`float`):
+            The rho parameter used for calculating the Karras sigma schedule, introduced in the EDM paper [2]. This was
+            set to 7.0 in the original implementation, which is also the original value suggested in the EDM paper.
+        clip_denoised (`bool`):
+            Whether to clip the denoised outputs to `(-1, 1)`. Defaults to `True`.
+        timesteps (`List` or `np.ndarray` or `torch.Tensor`, *optional*):
+            Optionally, an explicit timestep schedule can be specified. The timesteps are expected to be in increasing
+            order.
+    """
+
+    order = 1
+
+    @register_to_config
+    def __init__(
+        self,
+        num_train_timesteps: int = 40,
+        sigma_min: float = 0.002,
+        sigma_max: float = 80.0,
+        sigma_data: float = 0.5,
+        s_noise: float = 1.0,
+        rho: float = 7.0,
+        clip_denoised: bool = True,
+    ):
+        # standard deviation of the initial noise distribution
+        self.init_noise_sigma = sigma_max
+
+        # setable values
+        self.num_inference_steps = None
+        self.timesteps = torch.from_numpy(np.arange(0, num_train_timesteps)[::-1].copy())
+        self.custom_timesteps = False
+        self.is_scale_input_called = False
+
+    def index_for_timestep(self, timestep, schedule_timesteps=None):
+        if schedule_timesteps is None:
+            schedule_timesteps = self.timesteps
+
+        indices = (schedule_timesteps == timestep).nonzero()
+        return indices.item()
+
+    def scale_model_input(
+        self, sample: torch.FloatTensor, timestep: Union[float, torch.FloatTensor]
+    ) -> torch.FloatTensor:
+        """
+        Scales the consistency model input by `(sigma**2 + sigma_data**2) ** 0.5`, following the EDM model.
+
+        Args:
+            sample (`torch.FloatTensor`): input sample
+            timestep (`float` or `torch.FloatTensor`): the current timestep in the diffusion chain
+        Returns:
+            `torch.FloatTensor`: scaled input sample
+        """
+        # Get sigma corresponding to timestep
+        if isinstance(timestep, torch.Tensor):
+            timestep = timestep.to(self.timesteps.device)
+        step_idx = self.index_for_timestep(timestep)
+        sigma = self.sigmas[step_idx]
+
+        sample = sample / ((sigma**2 + self.config.sigma_data**2) ** 0.5)
+
+        self.is_scale_input_called = True
+        return sample
+
+    def scale_timestep(self, timestep: Union[float, torch.FloatTensor]):
```

Comment:
```suggestion
    def _scale_timestep(self, timestep: Union[float, torch.FloatTensor]):
```

Reply:
Let's scale all timesteps before the loop

Reply:
See https://github.com/huggingface/diffusers/pull/3492#discussion_r1246049934.